engine/src/distribution/MessageManager.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <vector>
+#include <map>
+#include "execution_graph/logic_controllers/CacheMachine.h"
+#include <blazingdb/manager/Context.h>
+
+namespace ral {
+namespace distribution {
+
+using Context = blazingdb::manager::Context;
+
+class MessageManager {
```

Comment:
This  class seems to make the assumption that a message is only destined for one place which is incorrect.

Reply:
Why do you say that I don't believe it makes that assumption at all